oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/OneandoneComputeServiceAdapter.java

```diff
@@ -0,0 +1,214 @@
+
+      try {
+         List<? extends Processor> processors = hardware.getProcessors();
+         org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware hardwareRequest
+                 = org.apache.jclouds.oneandone.rest.domain.Hardware.CreateHardware.create(processors.size(), cores, hardware.getRam(), hdds);
+         final Server.CreateServer serverRequest = Server.CreateServer.builder()
+                 .name(name)
+                 .hardware(hardwareRequest)
+                 .applianceId(image.getId())
+                 .dataCenterId(dataCenterId)
+                 .password(password)
+                 .powerOn(Boolean.TRUE).build();
+
+         logger.trace("<< provisioning server '%s'", serverRequest);
+
+         server = api.serverApi().create(serverRequest);
+         logger.trace(">> provisioning complete for server. returned id='%s'", server.id());
+
+      } catch (Exception ex) {
+         logger.error(ex, ">> failed to provision server. rollbacking..");
+         throw Throwables.propagate(ex);
+      }
+      waitServerUntilAvailable.apply(server);
+      
+      //add the firewall policy logic here
+
+      LoginCredentials serverCredentials = LoginCredentials.builder()
+              .user(loginUser)
+              .password(password)
+              .build();
+
+      return new NodeAndInitialCredentials<Server>(server, server.id(), serverCredentials);
+
+   }
+
+   @Override
+   public List<HardwareFlavour> listHardwareProfiles() {
+      return api.serverApi().listHardwareFlavours();
+   }
+
+   @Override
+   public Iterable<Image> listImages() {
+      return api.imageApi().list();
+   }
+
+   @Override
+   public Image getImage(String id) {
+      // try search images
+      logger.trace("<< searching for image with id=%s", id);
+      Image image = api.imageApi().get(id);
+      if (image != null) {
+         logger.trace(">> found image [%s].", image.name());
+         return image;
+      }
+      throw new ResourceNotFoundException("No image with id '" + id + "' was found");
+   }
+
+   @Override
+   public Iterable<Location> listLocations() {
+      // Will never be called
+      throw new UnsupportedOperationException("Locations are configured in jclouds properties");
+   }
+
+   @Override
+   public Server getNode(String id) {
+      return api.serverApi().get(id);
+   }
+
+   @Override
+   public void destroyNode(String id) {
+      waitServerUntilAvailable.apply(getNode(id));
+      logger.debug("Destroying %s ...", id);
+      api.serverApi().delete(id);
+      logger.debug("Destroyed %s!", id);
+   }
+
+   @Override
+   public void rebootNode(String id) {
+      waitServerUntilAvailable.apply(getNode(id));
+      api.serverApi().updateStatus(id, Server.UpdateStatus.create(Types.ServerAction.REBOOT, Types.ServerActionMethod.HARDWARE));
+   }
+
+   @Override
+   public void resumeNode(String id) {
+      waitServerUntilAvailable.apply(getNode(id));

```

Comment:
This won't work? You wait until the node is powered on when you want to resume it. This will take forever.